src/session_impl.cpp

```diff
@@ -630,10 +630,26 @@ namespace aux {
 		update_upnp();
 		update_natpmp();
 		update_lsd();
-		update_dht();
 		update_peer_fingerprint();
-		update_dht_bootstrap_nodes();
-		update_dht_announce_interval();
+
+#ifndef TORRENT_DISABLE_DHT
+		// setup DHT
+		if (m_outstanding_router_lookups == 0)
```

Comment:
I don't understand how ``session_impl::init()`` can be called more than once. Is ``m_outstanding_router_lookups`` ever non-zero?